### PR TITLE
Typing the typescript

### PR DESCRIPTION
**Pull request by hex2f**
Typing the typescript with types (by adding types)

### CODE REVIEW

File: components/Alert.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+function getClasses(type: AlertType) {
   return classNames("alert block p-4", {
     "bg-red-100 text-red-500": type === "danger",
     "bg-yellow-100 text-yellow-500": type === "warn",
     "bg-blue-100 text-blue-500": type === "info",
   });
 }
 
-export default function Alert({ type, children }) {
+const Alert: React.FC<{ type: AlertType }> = ({ type, children }) => {
```
Comment (IanMitchell) [inline components/Alert.tsx]: :megapuke: can this be done while keeping a `function Name()` syntax?

Comment (hex2f) [reply components/Alert.tsx]: Yup! That's just the way I usually do it. Changed it to use interfaces for defining prop types in my latest commit.

File: components/Header.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ export default function Header() {
         <Menu.Items className="absolute right-0 mt-2 w-56 bg-white rounded-md focus:outline-none shadow-lg origin-top-right ring-1 ring-black ring-opacity-5">
           <div className="py-1">
             <Menu.Item>
-              {({ active }) => (
+              {({ active }: { active: boolean }) => (
```
Comment (IanMitchell) [inline components/Header.tsx]: these are already typed

File: components/MDX.tsx [hunk 3]
```diff
@@ -3,22 +3,28 @@ import ContentWrapper from "./mdx/ContentWrapper";
 import Heading from "./mdx/Heading";
 import Code from "./mdx/Code";
 
-function InlineCode(props) {
+const InlineCode: React.FC<{}> = (props) => {
   return <code {...props} />;
 }
 
+type HeadingProps = React.DetailedHTMLProps<React.HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>;
```
Comment (IanMitchell) [inline components/MDX.tsx]: this should be in the heading file

**Pull request merged**